src/sentry/tasks/collect_project_platforms.py

```diff
@@ -37,5 +31,32 @@ def collect_project_platforms(**kwargs):
             )
         min_project_id += step
 
+
+@instrumented_task(name="sentry.tasks.collect_project_platforms", queue="stats")
+def collect_project_platforms(**kwargs):
+    now = timezone.now()
+
+    pre_snowflake_max_value = 2147483647
+
+    min_project_id = 0
+    max_project_id_before_snowflake = Project.objects.filter(
+        id__lt=pre_snowflake_max_value
+    ).aggregate(x=Max("id"))["x"]
+    step = 1000
+
+    _collect_project_platforms(min_project_id, max_project_id_before_snowflake, now, step)
+
+    min_project_id_after_snowflake = Project.objects.filter(
+        id__gt=pre_snowflake_max_value
+    ).aggregate(x=Min("id"))["x"]
+    max_project_id_after_snowflake = Project.objects.filter(
+        id__gt=pre_snowflake_max_value
+    ).aggregate(x=Max("id"))["x"]
+    step = max((max_project_id_after_snowflake - min_project_id_after_snowflake) / 1000, 1)
+
+    _collect_project_platforms(
+        min_project_id_after_snowflake, max_project_id_after_snowflake, now, step
+    )
```

Comment:
I could use some clarification on what the intent is here. As far as I can tell from the previous code, the intent was for `max_project_id` to encompass all projects in the database, and to iterate over them in "steps" of 1000 projects each. So we're *not* relying on monotonically increasing IDs to tell us when any particular slice of rows were created, right? (If we were, I'd be talking about scrapping that entire approach and rewriting this to use timestamps.)

However, even if we don't need to worry about whether Snowflake IDs are monotonic, I don't think this "step" logic is going to work with non-sequential IDs. If I'm understanding the intent correctly, the old approach exploited the property that ID values were never skipped, so a range of 1000 ID values would pretty much encompass 1000 rows (barring deletions and other minor details). But the timestamp part of our Snowflake ID scheme increments by 65536 once each second, which means so will the value of `max_project_id_after_snowflake`. I believe this means the number of projects in each "step" decreases dramatically compared to pre-Snowflake, and the total number of steps increases by 65 for every second that passes after the Snowflake epoch? That's no good.

I think we need to abandon the approach of manually slicing the `Project` table up by ID ranges. If we want to limit the `Group` queries to 1000 projects at a time, we should just make paginated queries over the `Project` table and fetch 1000 IDs at a time. That will make this overall operation more expensive (where the original hack with upper and lower ID bounds needed only O(1) `Project` reads to get those bounds), but such is the price of distributed architecture, I suppose.